### DISCUSSION TITLE
Highlight Code Blocks in git log's commit message.

### DISCUSSION THREAD

**Discussion by eval-exec** [magit/magit#5054] [Ideas]
I think it would be nice to support markdown syntax highlight in commit message:

For example, in `magit-log-mode`:
The `DescriptorScriptPubKeyMan` have no color highlight:
![image](https://github.com/magit/magit/assets/46400566/cf7937dd-4ea7-4e97-8e4a-6fc944de8883)

But in github's web page, the code in commit message have markdown highlight, show as mono font with gray background:
![image](https://github.com/magit/magit/assets/46400566/e7acc3dd-3d7d-44d3-abb4-4be5ff640cea)

I think it would be more cool of magit can support this.

**Comment by tarsius** [comment]
You can use `magit-log-format-message-function`. It's undocumented but looking at how it is used and how the function used as its default value works, should be enough to figure things out. (Though I probably should document it.)

I haven't added this because of the performance implications and because different people use different styles; I, for example, use symbol-in-elisp-docstring quoting instead of markdown quoting.

**Comment by eval-exec** [comment]
Thank you!
